### PR TITLE
Add a simple HTTP resolver

### LINKED ISSUE #6920: Add URL Resolver
### Feature request

Would be nice to add URL Resolver, that fetches Task/Pipeline directly from the URL. Pipelines as Code support [this kind of resolver](https://pipelinesascode.com/docs/guide/resolver/#remote-http-url).

### Use case

It useful to reference a task that is hosted on some remote URL or even in a git repo. Using Git Resolver on a large repository is not convenient, becase it takes a lot of time to fetch the whole repo and the use the file. If we can just reference the file by URL, the resolving will be much faster.

### PR DESCRIPTION
**Pull request by chmouel**
This adds a simple HTTP resolver that can fetch a file from a remote HTTP(S) URL.

Only fetch timeout configuration for now is supported.

This is kept simple for now, and does not support any kind of HTTP authentication, custom TLS or any other features. Something we can improve on later.

Fixes #6920 

# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [X] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [X] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [X] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [X] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [X] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
A new HTTP resolver is now available, it will let you resolve a http(s) URL to fetch a task or pipeline from.
```

/kind feature

### CODE REVIEW

File: docs/http-resolver.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: remote-task-reference
+spec:
+  taskRef:
+    resolver: http
+    params:
+    - name: url
+    value: https://raw.githubusercontent.com/tektoncd-catalog/git-clone/main/task/git-clone/git-clone.yaml
```
Comment (jerop) [inline docs/http-resolver.md:54]:
```suggestion
    params:
    - name: url
      value: https://raw.githubusercontent.com/tektoncd-catalog/git-clone/main/task/git-clone/git-clone.yaml
```

File: pkg/resolution/resolver/http/resolver.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+	httpResolverName = "Http"
+
+	// ConfigMapName is the http resolver's config map
+	configMapName = "http-resolver-config"
+
+	// default Timeout value when fetching http resources in seconds
+	defaultHttpTimeoutValue = "1m"
+)
+
+// Resolver implements a framework.Resolver that can fetch files from OCI bundles.
```
Comment (Yongxuanzhang) [inline pkg/resolution/resolver/http/resolver.go:51]: is this related to OCI bundles?

File: pkg/resolution/resolver/http/resolver.go [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+// GetSelector returns a map of labels to match requests to this resolver.
+func (r *Resolver) GetSelector(context.Context) map[string]string {
+	return map[string]string{
+		common.LabelKeyResolverType: LabelValueHttpResolverType,
+	}
+}
+
+// ValidateParams ensures parameters from a request are as expected.
+func (r *Resolver) ValidateParams(ctx context.Context, oParams []pipelinev1.Param) error {
```
Comment (Yongxuanzhang) [inline pkg/resolution/resolver/http/resolver.go:76]: nit: why this is `oParams`?

**Review by Yongxuanzhang — COMMENTED** [review]
Thanks! Just have some minor questions.
Shall we create an issue to track potential future work as you mentioned in the pr description?

**Comment by chmouel** [comment]
> Shall we create an issue to track potential future work as you mentioned in the pr description?

thanks for the review, yes i am planning to fill/tackle those issues when this is merged !

File: pkg/resolution/resolver/http/resolver.go (hunk 3 above)
Comment (chmouel) [reply pkg/resolution/resolver/http/resolver.go:76]: ah yeah there was two set of params at first in that function the map[string]string  and the object ones so called differently, i refactored those things before submitting it and kept the old name...

**Review by vdemeester — APPROVED** [review]

**Comment by Yongxuanzhang** [comment]
Is it possible that we add an example as @jerop suggested?

**Comment by chmouel** [comment]
@Yongxuanzhang I didn't see @jerop comment, thanks!

**Comment by Yongxuanzhang** [comment]
> @Yongxuanzhang I didn't see @jerop comment, thanks!
> 
> I just pushed it!

Thanks!
/lgtm

**Comment by chmouel** [comment]
The CI issue was some yaml whitespace errors 🍤
